src/Component/Navbar.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useMemo, useState, useCallback } from "react";
+import "../App.css";
+import CurrencyData from "../data/currency-codes.json";
+import LanguageData from "../data/language-codes.json";
+import RegionData from "../data/regions.json";
+
+export const Navbar: React.FC = () => {
+  // State variables to track selected currency, language, region, and search query
+  const [selectedCurrency, setSelectedCurrency] = useState<string>("");
+  const [selectedLanguage, setSelectedLanguage] = useState<string>("");
+  const [selectedRegion, setSelectedRegion] = useState<string>("");
+  const [search, setSearch] = useState<string>("");
+
+  // Memoized options for currency selection
+  const CurrencyOptions = useMemo(() => {
+    return (
+      <>
+        <option value="any">Any Currency </option>
+        {Object.entries(CurrencyData).map(([key, value]) => (
```

Comment:
Also, even if you were to put these in a useMemo, `CurrencyData` should be in the dependency aray. -1